### PR TITLE
feat(lambda-event-sources): disable event source

### LINKED ISSUE #5750: Lambda event sources can't be added as disabled when using add_event_source method
I'd like to add event sources to my lambda functions and have them be disabled on creation. This is currently not possible with the `add_event_source` method.

### Use Case

It should be possible to create event sources set to disabled.

### Proposed Solution

Currently you add event sources like this:

```python
my_lambda_function.add_event_source(
    SqsEventSource(my_queue)
)
```

An optional argument could be added that will be passed down to the `EventSourceMapping` that is created by the method:

```python
my_lambda_function.add_event_source(
    SqsEventSource(my_queue),
    enabled=True
)
```

Same should be done for the other arguments defined in `EventSourceMappingOptions`:
* `batch_size`
* `max_batching_window`
* `starting_position`

### Other

This might be a good first issue candidate.

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by zeynepsu**
One concern may be that there are a lot of event source mapping options specific to streams that are not reflected in the CDK code base: https://docs.aws.amazon.com/lambda/latest/dg/API_CreateEventSourceMapping.html. If perceived to be needed ASAP, those can be added to the StreamEventSourceProps in this CR as well. I checked that CloudFormation does support all of these options. However, only passing the enabled flag helps close the feature request #5750 and limits the scope of the CR. 
Unit tests are added and verification was added to existing integration tests as well. I also manually verified that all 3 event sources that support this (dynamo, sqs and kinesis) start out as disabled.

closes #5750

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by nija-at** [comment]
> ne concern may be that there are a lot of event source mapping options specific to streams that are not reflected in the CDK code base: https://docs.aws.amazon.com/lambda/latest/dg/API_CreateEventSourceMapping.html. If perceived to be needed ASAP, those can be added to the StreamEventSourceProps in this CR as well. I checked that CloudFormation does support all of these options. However, only passing the enabled flag helps close the feature request #5750 and limits the scope of the CR.

This is fine. We have another PR that addresses the other properties coming in - https://github.com/aws/aws-cdk/pull/5929

File: packages/@aws-cdk/aws-lambda-event-sources/README.md [hunk 1]
```diff
@@ -137,6 +138,7 @@ and add it to your Lambda function. The following parameters will impact Amazon
 
 * __batchSize__: Determines how many records are buffered before invoking your lambda function - could impact your function's memory usage (if too high) and ability to keep up with incoming data velocity (if too low).
 * __startingPosition__: Will determine where to being consumption, either at the most recent ('LATEST') record or the oldest record ('TRIM_HORIZON'). 'TRIM_HORIZON' will ensure you process all available data, while 'LATEST' will ignore all records that arrived prior to attaching the event source.
+* __enabled__: Disables or enables the event source mapping to pause/start polling and invocation.
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda-event-sources/README.md]: Is this not applicable to other event sources, such as S3 and SNS?

The [documentation here](https://docs.aws.amazon.com/lambda/latest/dg/API_CreateEventSourceMapping.html) seems to suggest that they're all event sources, no?

File: packages/@aws-cdk/aws-lambda-event-sources/test/integ.dynamodb.ts [hunk 2]
```diff
@@ -20,7 +20,8 @@ class DynamoEventSourceTest extends cdk.Stack {
 
     fn.addEventSource(new DynamoEventSource(queue, {
       batchSize: 5,
-      startingPosition: lambda.StartingPosition.TRIM_HORIZON
+      startingPosition: lambda.StartingPosition.TRIM_HORIZON,
+      enabled: false
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda-event-sources/test/integ.dynamodb.ts]: Avoid changing an existing integration test. 

Actually, an integration test is not needed for this feature; unit tests are sufficient.

**Comment by Gtofig** [comment]
Is this still being worked on?

**Pull request closed without merging**